### DISCUSSION TITLE
How to disable spout?

### DISCUSSION THREAD

**Discussion by Atque** [Stellarium/stellarium#1537] [Q&A] [answered]
Running the latest public beta build, we see the effect of commit 1bfb528 (#1507). This means that multisampling is disabled.

I can't find a way to disable spout, so that multisampling is enabled. When building from source, you can choose to disable spout in the configurations settings. Is there a way to run the beta build with spout disabled?

**Comment by gzotti** [comment, marked as answer]
There is no way to DISable spout. It is ENabled when called by setting the spout property using the `--spout=sky` command line argument.  If 

`qApp->property("spout").toString() == ""
`
does not provide true if spout is not set, this test must be changed. What is the value of `qApp->property("spout").toString()`?

**Reply by Atque** [reply]
> What is the value of qApp->property("spout").toString()?

How do I check that?

**Reply by gzotti** [reply]
Add this in the area where we decide upon using multisampling:

`qDebug() << "Value of 'spout':" <<  qApp->property("spout").toString();`

Then run and check log output. If the value of the 'spout' qApp property is not an empty string when not starting with --spout on the command line, we must change the test.

**Reply by gzotti** [reply]
Maybe this is better:

`if(  multisamplingLevel  && (!qApp->property("spout").isValid()) )`

**Reply by alex-w** [reply]
Why `isValid()`? Maybe `isEmpty()`?

**Reply by gzotti** [reply]
`isEmpty()` and `==""` should be equivalent. But I saw it today in the Qt helpfiles:

> QVariant QObject::property(const char *name) const
> Returns the value of the object's name property.
> If no such property exists, the returned variant is invalid.

If the --spout arg is not given on the commandline, the property may be invalid. I am fixing sth else now, though.

**Reply by gzotti** [reply]
Indeed 
`qDebug() << "Value of 'spout':" << qApp->property("spout").toString();`
gives

> Value of 'spout': "none"

and not an empty string. 

However
`qDebug() << "Value of 'spout' valid?:" << qApp->property("spout").isValid();`
gives

> Value of 'spout' valid?: true

and therefore the solution is
```
if(  multisamplingLevel  && (qApp->property("spout").toString() == "none") )
		fmt.setSamples(multisamplingLevel);
```

**Reply by gzotti** [reply]
Alright, done in 518ff19720c90.
Thanks @Atque for noticing the problem.

**Reply by Atque** [reply]
@gzotti That works, thanks!
